flink-tests/src/test/java/org/apache/flink/api/datastream/DataStreamBatchExecutionITCase.java

```diff
@@ -180,6 +207,350 @@ public void batchSumSingleResultPerKey() throws Exception {
+        }
+    }
+
+    @Test
+    public void batchMixedKeyedAndNonKeyedMultiInputOperator() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setRuntimeMode(RuntimeExecutionMode.BATCH);
+
+        DataStream<Tuple2<String, Integer>> bc1Input =
+                env.fromElements(Tuple2.of("bc3", 3), Tuple2.of("bc2", 2))
+                        .assignTimestampsAndWatermarks(
+                                WatermarkStrategy.<Tuple2<String, Integer>>forMonotonousTimestamps()
+                                        .withTimestampAssigner((in, ts) -> in.f1))
+                        .broadcast();
+        DataStream<Tuple2<String, Integer>> bc2Input =
+                env.fromElements(Tuple2.of("bc1", 1))
+                        .assignTimestampsAndWatermarks(
+                                WatermarkStrategy.<Tuple2<String, Integer>>forMonotonousTimestamps()
+                                        .withTimestampAssigner((in, ts) -> in.f1))
+                        .broadcast();
+        DataStream<Tuple2<String, Integer>> regularInput =
+                env.fromElements(
+                                Tuple2.of("regular1", 1),
+                                Tuple2.of("regular1", 2),
+                                Tuple2.of("regular1", 3),
+                                Tuple2.of("regular1", 4),
+                                Tuple2.of("regular2", 3),
+                                Tuple2.of("regular2", 5),
+                                Tuple2.of("regular1", 3))
+                        .assignTimestampsAndWatermarks(
+                                WatermarkStrategy.<Tuple2<String, Integer>>forMonotonousTimestamps()
+                                        .withTimestampAssigner((in, ts) -> in.f1))
+                        .keyBy(input -> input.f0);
+
+        KeyedMultipleInputTransformation<String> multipleInputTransformation =
+                new KeyedMultipleInputTransformation<>(
+                        "operator",
+                        mixedInputsOperatorFactory,
+                        BasicTypeInfo.STRING_TYPE_INFO,
+                        1,
+                        BasicTypeInfo.STRING_TYPE_INFO);
+        multipleInputTransformation.addInput(
+                regularInput.getTransformation(), input -> ((Tuple2<String, Integer>) input).f0);
+        multipleInputTransformation.addInput(bc1Input.getTransformation(), null);
+        multipleInputTransformation.addInput(bc2Input.getTransformation(), null);
+
+        DataStream<String> result =
+                new MultipleConnectedStreams(env).transform(multipleInputTransformation);
+
+        try (CloseableIterator<String> resultIterator = result.executeAndCollect()) {
+            List<String> results = CollectionUtil.iteratorToList(resultIterator);
+            // regular, that is non-keyed input is not sorted by timestamp. For keyed inputs
```

Comment:
Actually the non-broadcasted input is keyed, so we should probably remove the comment.

I copied it over from one of your tests and forgot to remove the comment. The same applies for the other manually crafted two input broadcast operator.